src/gwt/src/org/rstudio/studio/client/common/filetypes/TextFileType.java

```diff
@@ -450,8 +450,13 @@ public TokenPredicate getTokenPredicate()
          @Override
          public boolean test(Token token, int row, int column)
          {
-            return reTextType_.match(token.getType(), 0) != null
-                  && reNospellType_.match(token.getType(), 0) == null;
+            if (reNospellType_.match(token.getType(), 0) != null) {
+               return false;
+            }
```

Comment:
I prefer to separate negative and positive invariants when possible, especially when they're so similar as I find that it greatly improves code legibility. If this is strongly discouraged for some reason I'm happy to discuss and revert if necessary.

Reply:
FWIW I agree 100% (I try to exit early for anything not on the 'main' code path); but I do think some other team members might prefer having only a single point of return in a function (ie, at the end).

We don't have any hard rules here though -- I think whatever keeps the code legible and correct is fine.